src/libusermode/running.cpp

```diff
@@ -0,0 +1,490 @@
+ *                                                                         *
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include <fstream>
+#include <sstream>
+#include <map>
+#include <string>
+#include <optional>
+
+#include <config.h>
+#include <glib.h>
+#include <inttypes.h>
+#include <libvmi/libvmi.h>
+#include <libvmi/peparse.h>
+#include <libdrakvuf/libdrakvuf.h>
+#include <assert.h>
+
+#include "userhook.hpp"
+#include "uh-private.hpp"
+
+
+struct er_data_t
+{
+    // Arguments provided by the user.
+    addr_t target_process;
+    std::string dll_name;
+    std::string func_name;
+    callback_t cb;
+    void* extra;
+
+    // Additional data. Stored here for optimalization.
+    target_hook_state state;
+    vmi_pid_t target_process_pid;
+    addr_t target_process_dtb;
+    addr_t func_addr;
```

Comment:
This is doubling a lot of logic, but I can merge it later down the line with my refactor :)